cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,639 @@
+        """Returns a sequence of tuple pairs with the first item being a Rabi
+        angle and the second item being the corresponding excited state
+        probability.
+        """
+        return [(angle, prob) for angle, prob in zip(self._rabi_angles,
+                                                     self._excited_state_probs)]
+
+    def plot(self) -> None:
+        """Plots excited state probability vs the Rabi angle (angle of rotation
+        around the x-axis).
+        """
+        fig = pyplot.figure()
+        pyplot.plot(self._rabi_angles, self._excited_state_probs, 'ro-',
+                    figure=fig)
+        pyplot.xlabel(r"Rabi Angle ($\pi$)")
+        pyplot.ylabel('Excited State Probability')
+
+
+class RBResult:
+    """Results from a randomized benchmarking experiment."""
+
+    def __init__(self, numbers_cliffords: Sequence[int],
+                 ground_state_probabilities: Sequence[float]):
+        """
+        Args:
+            numbers_cliffords: The different numbers of Cliffords in the RB
+                study.
+            ground_state_probabilities: The corresponding average ground state
+                probabilities.
+        """
+        self._num_cfds_seq = numbers_cliffords
+        self._gnd_state_probs = ground_state_probabilities
+
+    @property
+    def data(self) -> Sequence[Tuple[int, float]]:
+        """Returns a sequence of tuple pairs with the first item being a
+        number of Cliffords and the second item being the corresponding average
+        ground state probability.
+        """
+        return [(num, prob) for num, prob in zip(self._num_cfds_seq,
+                                                 self._gnd_state_probs)]
+
+    def plot(self) -> None:
+        """Plots the average ground state probability vs the number of
+        Cliffords in the RB study.
+        """
+        fig = pyplot.figure()
+        pyplot.plot(self._num_cfds_seq, self._gnd_state_probs, 'ro-',
+                    figure=fig)
+        pyplot.xlabel(r"Number of Cliffords")
+        pyplot.ylabel('Ground State Probability')
+
+
+class TomographyResult:
+    """Results from a state tomography experiment."""
+
+    def __init__(self, density_matrix: np.ndarray):
+        """
+        Args:
+            density_matrix: The density matrix obtained from tomography.
+        """
+        self._density_matrix = density_matrix
+
+    @property
+    def data(self) -> np.ndarray:
+        """Returns an n^2 by n^2 complex matrix representing the density
+        matrix of the n-qubit system.
+        """
+        return self._density_matrix
+
+    def plot(self) -> None:
+        """Plots the real and imaginary parts of the density matrix as two
+        3D bar plots.
+        """
+        _plot_density_matrix(self._density_matrix)
+
+
+def rabi_oscillations(sampler: sim.SimulatesSamples, qubit: devices.GridQubit,
+                      final_angle: float, number_shots: int,
```

Comment:
Final angle needs to be in radians. We can convert it internally, but people running rabi oscillation experiments will be thinking in radians instead of powers of Paulis.

Final angle should default to pi or 2pi; your choice. Maybe rename to `max_angle`.

Is there a reasonable default we could use for the number of repetitions or the number of points? E.g. one thousand repetitions and one hundred angles?

Force keyword parameters for shots and points by adding a `, *,` parameter before them, so that it isn't unclear what. Rename them to `repetitions` and `num_points` for better consistency with other methods in cirq.

Reply:
All done.